x-pack/plugins/apm/server/lib/service_map/get_service_anomalies.ts

```diff
@@ -153,7 +161,7 @@ export async function getMLJobIds(
       (job) => job.custom_settings?.job_tags?.environment === environment
     );
     if (!matchingMLJob) {
-      throw new Error(`ML job Not Found for environment "${environment}".`);
+      return [];
```

Comment:
Changed this to return an empty array to prevent us from having to catch this error in various places.

Reply:
I think that's fine. Should we log this or is that just noisy?

Reply:
I think it will be noisy, I wouldn't consider a 404 an exception or even a warning.